src/db/database.py

```diff
@@ -0,0 +1,15 @@
+from sqlalchemy.ext.asyncio import create_async_engine, AsyncSession
+from sqlalchemy.orm import sessionmaker
+from src.db.model import Base
+
+class Database:
+    def __init__(self, database_url: str):
+        self.engine = create_async_engine(database_url, echo=False, future=True)
+        self.async_session = sessionmaker(self.engine, class_=AsyncSession, expire_on_commit=False)
```

Comment:
Try making the Database into something close to what Alex is doing for context management. Maybe think about adding the rollbacks in that context manager.